src/main/java/org/folio/rest/impl/ItemDamagedStatusAPI.java

```diff
@@ -0,0 +1,298 @@
+          .setHandler(asyncResultHandler);
+      } catch (Exception ex) {
+        LOGGER.error(ex.getMessage(), ex);
+        String message = messages.getMessage(lang, InternalServerError);
+        Response response = GetItemDamagedStatusesResponse.respond500WithTextPlain(message);
+        asyncResultHandler.handle(succeededFuture(response));
+      }
+    });
+  }
+
+  private Future<ItemDamageStatuses> searchItemDamagedStatuses(
+    String query,
+    int offset,
+    int limit,
+    Map<String, String> okapiHeaders,
+    Context vertxContext) throws FieldException {
+
+    Future<Results<ItemDamageStatus>> future = Future.future();
+
+    postgresClient(vertxContext, okapiHeaders)
+      .get(
+        REFERENCE_TABLE,
+        ItemDamageStatus.class,
+        new String[]{"*"},
+        getCQL(query, limit, offset, REFERENCE_TABLE),
+        true,
+        true,
+        future.completer()
+      );
+
+    return future.map(results ->
+      new ItemDamageStatuses()
+        .withItemDamageStatuses(results.getResults())
+        .withTotalRecords(results.getResultInfo().getTotalRecords())
+    );
+  }
+
+  @Override
+  public void postItemDamagedStatuses(
+    String lang,
+    ItemDamageStatus entity,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    vertxContext.runOnContext(v -> {
+      try {
+        saveItemDamagedStatus(entity, okapiHeaders, vertxContext)
+          .map(it -> respond201WithApplicationJson(it, headersFor201().withLocation(LOCATION_PREFIX + it.getId())))
+          .otherwise(ex ->
+            ofNullable(PgExceptionUtil.badRequestMessage(ex))
+              .map(PostItemDamagedStatusesResponse::respond400WithTextPlain)
+              .orElseGet(() -> respond500WithTextPlain(messages.getMessage(lang, InternalServerError)))
+          )
+          .map(Response.class::cast)
+          .setHandler(asyncResultHandler);
+      } catch (Exception e) {
+        LOGGER.error(e.getMessage(), e);
+        String message = messages.getMessage(lang, InternalServerError);
+        asyncResultHandler.handle(succeededFuture(respond500WithTextPlain(message)));
+      }
+    });
+  }
+
+  private Future<ItemDamageStatus> saveItemDamagedStatus(ItemDamageStatus entity, Map<String, String> okapiHeaders, Context vertxContext) {
+    Future<String> future = Future.future();
+    if (isNull(entity.getId())) {
+      entity.setId(UUID.randomUUID().toString());
+    }
+    postgresClient(vertxContext, okapiHeaders).save(REFERENCE_TABLE, entity.getId(), entity, future.completer());
+    return future.map(entity::withId);
+  }
+
+  @Override
+  public void getItemDamagedStatusesById(
+    String id,
+    String lang,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    vertxContext.runOnContext(v -> {
+      try {
+        getItemDamagedStatus(id, okapiHeaders, vertxContext)
```

Comment:
This approach is rather different to much of the existing code in the module, some of which has been updated to use the PgUtil class from RAML Module Builder.

Can you help me understand why this approach was chosen? Did it come from your work on another team?

Reply:
PgUtil uses the reflection API that is slow that's why I don't want to use it.
I don't think that this approach is completely different to existing code. I just move DB work out of controller to a method. 
Is that a problem? Shall I rewrite it using single method style?

Reply:
> PgUtil uses the reflection API that is slow that's why I don't want to use it.

I can understand those concerns. I imagine this is something you or colleagues discovered on another team. Have you or the other team, provided this feedback to the core platform team?

Where possible, I think it is better if we can improve the shared tooling (admittedly, mod-circulation is the biggest outlier in the project in this regard)

(Aside: A long time ago I tried to write a set of classes for making storage endpoints relatively easy to write. However, at the time I didn't feel I could achieve it whilst sticking with the idioms of RAML Module Builder. 

I subsequently abandoned my attempts as I worked more on business logic modules and the PgUtil was introduced.)

> I don't think that this approach is completely different to existing code.

I may well have missed that, I'm not too familiar with the storage modules anymore. Can you refer me to examples of existing code like this?

> Is that a problem? Shall I rewrite it using single method style?

At this point, my goal is to understand. It likely isn't my call as to whether to do that or not.

Reply:
> I can understand those concerns. I imagine this is something you or colleagues discovered on another team. Have you or the other team, provided this feedback to the core platform team?

Do you ask me to say core platform guys that PgUtil is slow and they need to create a ticket to rewrite it without reflection API?

> Can you refer me to examples of existing code like this?

Here is similar code [DataImportImpl.java](https://github.com/folio-org/mod-data-import/blob/2d579b9b054fbbc0bc97f3e8c6a0b2b2e633490d/src/main/java/org/folio/rest/impl/DataImportImpl.java#L94)

Reply:
> > I can understand those concerns. I imagine this is something you or colleagues discovered on another team. Have you or the other team, provided this feedback to the core platform team?
> 
> Do you ask me to say core platform guys that PgUtil is slow and they need to create a ticket to rewrite it without reflection API?
> 

I'm suggesting that if you, and others, are rejecting the use of PgUtil, then I think it could be worth giving feedback as to why to the core platform team.

Otherwise, we likely end up with another significantly different approach within the storage modules.

> > Can you refer me to examples of existing code like this?
> 
> Here is similar code [DataImportImpl.java](https://github.com/folio-org/mod-data-import/blob/2d579b9b054fbbc0bc97f3e8c6a0b2b2e633490d/src/main/java/org/folio/rest/impl/DataImportImpl.java#L94)

Thank you for the example. 

Do you know if that team chose to do this for the same reason?

Reply:
As discussed on Slack, this is a broader topic and should be discussed with a broader audience on a JIRA issue.

Given that, I'm shall leave it to other reviewers if they want to continue this topic within this pull request.